File: 1885-count-number-of-homogenous-substrings/1885-count-number-of-homogenous-substrings.cpp
class Solution {
public:
    int countHomogenous(string s) {
        const long long mod = 1e9+7;
        long long total_cnt, length, cnt;
        char prev_char;
        length = s.size();
        prev_char = s[0];
        total_cnt = 0;
        cnt = 0;
        for (int i=0; i<length; i++){
            if (s[i] == prev_char){
                cnt++;
            } else{
                for (int j=1; j<=cnt; j++){
                    total_cnt += j;
                }
                cnt=1;
            }
            prev_char = s[i];
        }
        for (int j=1; j<=cnt; j++){
            total_cnt += j;
        }
        return total_cnt%mod;
    }
};

File: 1885-count-number-of-homogenous-substrings/1885-count-number-of-homogenous-substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        total_cnt = 0
        prev_char = s[0]
        cnt = 0
        for char in s:
            if char == prev_char:
                cnt += 1
            else:
                total_cnt += sum(range(cnt+1))
                cnt = 1
            prev_char = char
        
        cnt+=1
        total_cnt += sum(range(cnt))
        
        return total_cnt % (10**9 + 7)

